src/client/datascience/notebookStorage/vscNotebookModel.ts

```diff
@@ -126,28 +117,30 @@ export class VSCodeNotebookModel extends BaseNotebookModel {
     public associateNotebookDocument(document: NotebookDocument) {
         this.document = document;
     }
-    public trust() {
-        // this._doNotUseOldCells = true;
-        super.trust();
-        this._cells = [];
-    }
     public async trustNotebook() {
-        super.trust();
-        if (this.document) {
-            const editor = this.vscodeNotebook?.notebookEditors.find((item) => item.document === this.document);
-            if (editor) {
-                await updateVSCNotebookAfterTrustingNotebook(editor, this.document, this._cells);
-            }
-            // We don't need old cells.
-            this._cells = [];
-            this._trustedAfterOpeningNotebook = true;
+        this.trust();
+        const editor = this.vscodeNotebook?.notebookEditors.find((item) => item.document === this.document);
+        const document = editor?.document;
+        if (editor && document && !document.metadata.trusted) {
```

Comment:
When trusting all we now need to do is, just update the notebook document meteadata